drivers/accel/adxl38x/adxl38x.c

```diff
@@ -0,0 +1,957 @@
+ * @param y    		- Y-axis's output data.
+ * @param z    		- Z-axis's output data.
+ *
+ * @return ret 		- Result of the reading procedure.
+*******************************************************************************/
+int adxl38x_get_xyz_gees(struct adxl38x_dev *dev,
+			 enum adxl38x_ch_select channels,
+			 struct adxl38x_fractional_val *x, struct adxl38x_fractional_val *y,
+			 struct adxl38x_fractional_val *z)
+{
+	int ret;
+	uint16_t raw_accel_x;
+	uint16_t raw_accel_y;
+	uint16_t raw_accel_z;
+	uint16_t raw_accel_t;
+
+	ret = adxl38x_get_raw_data(dev, channels, &raw_accel_x, &raw_accel_y,
+				   &raw_accel_z, &raw_accel_t);
+	if (ret)
+		return ret;
+
+	x->integer = no_os_div_s64_rem((int64_t)adxl38x_accel_conv(dev, raw_accel_x),
+				       ADXL38X_ACC_SCALE_FACTOR_GEE_DIV, &(x->fractional));
+	y->integer = no_os_div_s64_rem((int64_t)adxl38x_accel_conv(dev, raw_accel_y),
+				       ADXL38X_ACC_SCALE_FACTOR_GEE_DIV, &(y->fractional));
+	z->integer = no_os_div_s64_rem((int64_t)adxl38x_accel_conv(dev, raw_accel_z),
+				       ADXL38X_ACC_SCALE_FACTOR_GEE_DIV, &(z->fractional));
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Converts raw acceleration value to m/s^2 value.
+ *
+ * @param dev       - The device structure.
+ * @param raw_accel - Raw acceleration value.
+ *
+ * @return ret      - Converted data.
+*******************************************************************************/
+static int64_t adxl38x_accel_conv(struct adxl38x_dev *dev,
+				  uint16_t raw_accel)
+{
+	int32_t accel_data;
+
+	// Raw acceleration is in two's complement
+	// Convert from two's complement to int
+	if (raw_accel & NO_OS_BIT(15))
+		accel_data = raw_accel | ADXL38X_NEG_ACC_MSK;
+	else
+		accel_data = raw_accel;
+
+	// Apply scale factor based on the selected range
+	switch (dev->dev_type) {
+	case ID_ADXL380:
+		return ((int64_t)(accel_data * ADXL380_ACC_SCALE_FACTOR_GEE_MUL *
+				  adxl38x_scale_mul[dev->range]));
+	case ID_ADXL382:
+		return ((int64_t)(accel_data * ADXL382_ACC_SCALE_FACTOR_GEE_MUL *
+				  adxl38x_scale_mul[dev->range]));
+	default:
+		return -EINVAL;
+	}
+}
+
+/***************************************************************************//**
+ * @brief Puts the part in a safe state before setting important register values.
+ *
+ * @param dev       - The device structure.
+ *
+ * @return ret      - Result of setting to stanby mode.
+*******************************************************************************/
+static int adxl38x_set_to_standby(struct adxl38x_dev *dev)
+{
+	int ret;
+	uint8_t op_mode_reg_val;
+
+	ret = adxl38x_read_device_data(dev, ADXL38X_OP_MODE, 1, &op_mode_reg_val);
+	// Applying mask sets underlying op_mode to standby (0)
+	op_mode_reg_val = (op_mode_reg_val & ~ADXL38X_MASK_OP_MODE);
+	ret |= adxl38x_write_device_data(dev, ADXL38X_OP_MODE,
+					 1, &op_mode_reg_val);
+
+	// 10ms wait in the standby mode
```

Comment:
if this time is documented in datasheet please mention its name here

Reply:
Not mentioned. Was mentioned by coworker, but mostly unnecessary, so removed.